src/db.rs

```diff
@@ -215,30 +215,34 @@ impl DbInner {
 		self.columns[col as usize].get_size(&key, log)
 	}
 
-	// Commit is simply adds the the data to the queue and to the overlay and
+	// Commit simply adds the the data to the queue and to the overlay and
 	// exits as early as possible.
 	fn commit<I, K>(&self, tx: I) -> Result<()>
-		where
-			I: IntoIterator<Item=(ColId, K, Option<Value>)>,
-			K: AsRef<[u8]>,
+	where
+		I: IntoIterator<Item=(ColId, K, Option<Value>)>,
+		K: AsRef<[u8]>,
 	{
-		{
-			let bg_err = self.bg_err.lock();
-			if let Some(err) = &*bg_err {
-				return Err(Error::Background(err.clone()));
-			}
-		}
-
 		let commit: Vec<_> = tx.into_iter().map(
 			|(c, k, v)| (c, self.columns[c as usize].hash(k.as_ref()), v)
 		).collect();
 
+		self.commit_raw(commit)
+	}
+
+	fn commit_raw(&self, commit: Vec<(ColId, Key, Option<Value>)>) -> Result<()> {
```

Comment:
It could be done in the log writer thread. We want to make as little work in the `commit` function as possible, so that it returns as early as possible. 
I was considering changing the API from the `(column, key, Option<value>)` iterator to more of `(column, Action)` where `Action` is one of `Insert`, `Remove`, `ChangeRef`. We will need something like this for subindexes anyway.